portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -48,23 +59,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String comment = getParameter(request, ENTITY_COMMENT_HEADER, /* DefaultValue= */ "");
 
     // Respond with a refresh and do local and persistent storage updates.
-    comments.add(String.format("%s says: %s", name, comment));
-    storeComments(name, comment);
-    
+    storeComments(name, comment, timestamp);
     response.sendRedirect("/index.html");
   }
 
-  /**
-   * Converts a DataServlet instance into a JSON string using the Gson library.
-   */
+  private static String buildComment(Entity entity) {
+    String name = (String) entity.getProperty(ENTITY_NAME_HEADER);
+    String comment = (String) entity.getProperty(ENTITY_COMMENT_HEADER);
+    long timestamp = (long) entity.getProperty(ENTITY_TIMESTAMP_HEADER);
+    return getCommentStatement(name, comment, timestamp);
+  }
+
   private static String convertToJson(List<String> data) {
     return new Gson().toJson(data);
   }
 
-  /**
-   * @return the request parameter, or the default value if the parameter
-   *         was not specified by the client
-   */
+  private static String getCommentStatement(String name, String comment, long timestamp) {
+    return String.format("at %d, %s said: %s", timestamp, name, comment);
```

Comment:
you can just return this in buildComment
another way is you implement this as a toString() method in Comment class, and in buildComment you build a Comment object and return comment.toString().